src/main/java/uk/gov/hmcts/ccd/security/filters/LoggingFilter.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.hmcts.ccd.security.filters;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.web.filter.OncePerRequestFilter;
+import uk.gov.hmcts.ccd.data.SecurityUtils;
+import uk.gov.hmcts.reform.idam.client.models.UserInfo;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+@Slf4j
+public class LoggingFilter extends OncePerRequestFilter {
+
+    private static final String LOG_MESSAGE_TEMPLATE = "[LOG FILTER] Attempting to serve request %s %s for user with IDAM roles %s";
+
+    private final SecurityUtils securityUtils;
+    private final Pattern pathPattern;
+
+    public LoggingFilter(SecurityUtils securityUtils, String pathRegex) {
+        this.securityUtils = securityUtils;
+        this.pathPattern = Pattern.compile(pathRegex);
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+        throws ServletException, IOException {
+        log.debug(buildLogMessage(request, securityUtils.getUserInfo()));
```

Comment:
isn't it simpler to just log on INFO? it's definitely an important type of information that can be logged as INFO. 
This would be a safer approach as we wouldn't need to change the AppInsights log level